docs/api/DesktopAgent.md

```diff
@@ -231,9 +231,9 @@ Removes the app from any channel membership.  Context broadcast and listening th
 
 ```js
 //desktop-agent scope context listener
- const fdc3Listener = fdc3.addContextListener(context => {});
+const fdc3Listener = fdc3.addContextListener(context => {});
 
-fdc3.leaveCurrentChannel();
+await fdc3.leaveCurrentChannel();
```

Comment:
Why is this now asynchronous?

Reply:
I suggested that based on the fact that `joinChannel` being asynchronous. It could be useful to know when the leave operation successfully completed (or not).

Reply:
@vmehtafds @nkolba  I don't understand why this requires it to be asynchronous. success/error reporting is not the same thing as long-running operations that need to be asynchronous.

If you call the method without an exception, it worked. If it throws an exception, it didn't. If leaving (or joining) a channel as implemented in anything more then O(1) time, the implementation is doing something very weird.

I think making this operation asynchronous adds unnecessary complexity and ambiguity, and is a mistake.

Reply:
@rikoe  @vmehtafds . I'm in favor of keeping things simple.  
Given that joinChannel is async - I was thinking symmetry and consistency  (most other methods are async).  However, there is definitely a good case to be made for keeping it sync.

Reply:
Is joinChannel async? 🤔 I would argue the same applies. I do agree that both join and leave should both be consistent though...

Arguably both join and leaving channels should be instant no-ops, but happy to go with the majority view here.

Reply:
I could see wanting to do this: 
```js
await fdc3.joinChannel('global');
fdc3.broadcast(context);
```